src/components/CreateNewRocket.svelte

```diff
@@ -46,27 +100,66 @@
 		userHasProfile = userHasProfile;
 	}
 
+	let name: string = '';
+
+	$: nameInvalid = true;
+	$: nameError = '';
+	$: needBuyName = true;
+	$: canBuyName = true;
+	$: buyNameError = '';
+	$: cannotPublish = true;
+
 	const rocketNameValidator = /^\w{4,20}$/;
-	const nameIsUnique = (name: string, rocketEvents: NDKEvent[]) => {
-		return !rocketEvents.some((event) => event.tags[0][1] === name);
+
+	const nameIsUnique = (name: string, rockets: Rocket[]) => {
+		return !rockets.some((r) => r.Name() === name);
 	};
 
-	$: if (name) {
-		if (!rocketNameValidator.test(name)) {
+	$: {
+		if (!name) {
+			nameInvalid = true;
+			nameError = '';
+		} else if (!rocketNameValidator.test(name)) {
 			nameInvalid = true;
 			nameError = 'Rocket names MUST be 4-20 alphanumeric characters';
-		} else if (!$rocketsStore) {
-			// nameInvalid = true;
-			// nameError = 'Loading Nostr';
-		} else if (!nameIsUnique(name, $rocketsStore)) {
-			// nameInvalid = true;
-			// nameError = 'Rocket names MUST be unique';
+		} else if ($rockets && !nameIsUnique(name, $rockets)) {
+			nameInvalid = true;
+			nameError = 'Rocket names MUST be unique';
 		} else {
 			nameInvalid = false;
 			nameError = '';
 		}
 	}
 
+	$: {
+		const isPurchasedByOthers =
+			$purchasedRocketNames.all.includes(name) && !$purchasedRocketNames.my.includes(name);
+		const isPurchasedByMe = $purchasedRocketNames.my.includes(name);
+
+		if (isPurchasedByOthers) {
+			canBuyName = false;
+			buyNameError = 'This name has already been purchased by someone else, you cannot buy it';
+		} else if (isPurchasedByMe) {
+			canBuyName = false;
+			buyNameError = 'This name does not need to be purchased again.';
+		} else {
+			canBuyName = true;
+			buyNameError = '';
+		}
+
+		needBuyName = !$purchasedRocketNames.all.includes(name);
+	}
+
+	$: {
+		if (!$mainnet && !nameInvalid) {
```

Comment:
we can stop using the old way of testing for mainnet/testnet now, and only consider paid rockets to be mainnet, everything else is testnet

Reply:
The only exception is nostrocket rocket itself which should be hard coded as mainnet. So right now, since no one has paid, only nostrocket should appear as mainnet

Reply:
I want to divide the mainnet and testnet sections on the page, which will be done in other PRs. This PR only handles the creation of the rocket part.